app/src/main/java/com/ecosystem/kin/app/model/SignInRepo.java

```diff
@@ -2,43 +2,132 @@
 
 import android.content.Context;
 import android.content.SharedPreferences;
-import com.ecosystem.kin.app.R;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+import android.util.Base64;
+import com.ecosystem.kin.app.BuildConfig;
 import com.kin.ecosystem.network.model.SignInData;
 import com.kin.ecosystem.network.model.SignInData.SignInTypeEnum;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.PrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Date;
 import java.util.UUID;
 
 public class SignInRepo {
 
     private final static String USER_PREFERENCE_FILE_KEY = "USER_PREFERENCE_FILE_KEY";
     private final static String USER_UUID_KEY = "USER_UUID_KEY";
     private static final String DEVICE_UUID_KEY = "DEVICE_UUID_KEY";
+    private static final long MILLISECONDS_IN_DAY = 1000 * 60 * 60 * 24;
```

Comment:
you can use: `TimeUnit.DAYS.toMillis(1);`
also i would rename to `DAY_IN_MILLISECONDS`